internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -346,9 +349,10 @@ func (r *BMACReconciler) handleBMHFinalizer(ctx context.Context, log logrus.Fiel
 		return reconcileComplete{}
 	}
 
-	// BMH is being deleted and finalizer is gone, ensure detached is gone and return
+	// BMH is being deleted and finalizer is gone, ensure detached and paused are gone and return
 	if !bmhHasFinalizer {
 		dirty := removeBMHDetachedAnnotation(log, bmh)
+		dirty = removeBMHPausedAnnotation(log, bmh) || dirty
```

Comment:
Is there any situation where we don't do both of these?
Would it make more sense to have a single function to "reactivate" a BMH?

Reply:
Yeah, good idea, we can just consolidate these.